core/src/main/java/hudson/model/TimeZoneProperty.java

```diff
@@ -105,7 +107,7 @@ public static String forCurrentUser() {
         }
 
         TimeZone tz = TimeZone.getTimeZone(tzp.timeZoneName);
-        if (tz.getID() != tzp.timeZoneName) {
+        if (!Objects.equals(tz.getID(), tzp.timeZoneName)) {
```

Comment:
Not sure whether it is a real defect. Unlikely, because this type should be using static or inlined values. But fixing it would not hurt for sure